### ISSUE TITLE
recoverable wallet - what happens if I lose / disclose my private key?

### ISSUE THREAD

**Issue by dckc** [dckc/awesome-ocap#20]
Are there ocap patterns that address risks around key management?

> If the answer to 'what happens if I lose my keys?' is 'crumple the car,' you're doing it wrong. -- @benadida (_I think; I'm struggling to confirm._)

The [recoverable wallet](https://github.com/Chia-Network/wallets/blob/master/docs/recoverable_wallet.md) in chialisp looks promising, though I have just started studying it in detail. I wonder if there's an analog using ocaps rather than crypto.

cc @erights @warner @michaelfig @chrishibbert @katelynsills
LMK if/when I should move this to an Agoric repo.

**Comment by katelynsills** [comment]
My understanding is that you'll need cryptography at the edges no matter what, so I think there can't be an ocap solution here.

**Comment by dckc** [comment]
Right; good point... so perhaps a better question is: how to integrate such a recoverable wallet with ocaps?

Is it straightforward to layer on top of Agoric's design for message passing to/from cosmos, or is something else needed? I gather cosmos has multi-sig support; does that work for this case? Or is there something irreducible about the way chialisp uses BLSSignature?

**Comment by dckc** [comment]
The link to https://github.com/Chia-Network/wallets/blob/master/docs/recoverable_wallet.md has gone 404.

https://github.com/nondejus/wallets/blob/master/docs/recoverable_wallet.md seems to be another copy: Latest commit 18dcb1e on Dec 4, 2019

abstract:
> A Chia recoverable wallet is a wallet whose funds can be recovered in the event that the wallet is lost by providing a recovery string to another wallet. In contrast to a BIP 39 backup seed, a Chia recovery string is a low security piece of information. A thief who finds the string can't use it to steal funds from a wallet they don't own. This is accomplished by the recovery process locking up the funds in escrow along with some mandatory staking funds. If the recovery process was initiated by an attacker, the attacker will lose the staking funds as the original wallet has the ability to clawback the escrow coins during the escrow period. If the recovery process was legitimate however, the owner can retrieve their funds from escrow at the end of the escrow period, along with the staking funds. In this demonstration the staking amount is 10% and the escrow period is 3 blocks.

I forked it as https://github.com/dckc/chialisp-wallets ; I'm not sure that makes it substantially more available, but it should make it easier to find.

**Comment by dckc** [comment]
related work:

 - [Cryptology ePrint Archive: Report 2021/289 \- Reactive Key\-Loss Protection in Blockchains](https://eprint.iacr.org/2021/289)
> We achieve this by a 3-phase Commit() -> Reveal() -> Claim() - or - Challenge() smart contract that enables accessing funds of addresses for which the spending key is not available.

**Comment by dckc** [comment]
https://me.hashkey.com/en

In theory, it's a wallet that supports recovery via advanced crypto threshold signatures (TSS) and multiparty computation (MPC). Android app has ~0 users, though, and I couldn't find tech details.

**Comment by dckc** [comment]
Sunny's talk at [osmocon](https://osmocon.com/) included a slide on "Flexible Key Management" including "Social Recovery Systems"

![IMG_20220609_100813808](https://user-images.githubusercontent.com/150986/173891259-3dbdf3c8-de40-47a4-90ac-671ddaab7a10.jpg)

**Comment by dckc** [comment]
> ### Wallets & Custody
> Seed phrase wallets move out of style in favor of more user-friendly custody solutions leveraging MPC and account abstraction
-- [The 2023 Interchain Thesis: Exploring the biggest trends and opportunities in Cosmos \| Interop](https://medium.com/interopvc/2023-interchain-cosmos-thesis-b89ece3e21f7) by [Sebastien Couture](https://medium.com/@seb3point0?source=post_page-----b89ece3e21f7--------------------------------)
Feb 2

**Comment by dckc** [comment]
wow... https://github.com/ZenGo-X/cosmos-thresh-wallet

**Comment by dckc** [comment]
https://github.com/ZenGo-X/thresh-sig-js [37f0faa](https://github.com/ZenGo-X/thresh-sig-js/commit/37f0faa6426c30e39e2e405fb52597102834d189)
[on Jan 19, 2021](https://github.com/ZenGo-X/thresh-sig-js/commit/37f0faa6426c30e39e2e405fb52597102834d189)
Git stats

**Comment by dckc** [comment]
https://www.joinfountain.xyz/ crossed my radar... doesn't use an ocap approach, but it would be interesting to know more about what policies are in demand.

**Comment by dckc** [comment]
[Hedera and Algorand partner to develop decentralized wallet recovery system](https://cointelegraph.com/news/hedera-and-algorand-partner-to-develop-decentralized-wallet-recovery-system?utm_medium=email&_hsmi=82429022&utm_content=82429022&utm_source=hs_email)
